.github/workflows/test.yaml

```diff
@@ -99,3 +99,8 @@ jobs:
 
       - name: Run tests
         run: ./e2e/test_${{ matrix.test-script }}.sh
+      
+      - name: Upload logs for debugging
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v4
+        path: e2e-output
```

Comment:
Do we have any data limits we need to be aware of? There might be some big wheels or sdists in that directory.

Reply:
I am not sure i think there is a total limit of 100 GB. Someone experimented and said that greater than 5 GB uploads might start breaking (but i think that is the size of the zip archive): https://github.com/actions/upload-artifact/issues/9

Reply:
Could you also the default retention period of artifacts on this repository to be 7 days. It should be somewhere in the repo settings